pkg/services/app/add.go

```diff
@@ -45,6 +45,7 @@ type AddParams struct {
 }
 
 func (a *App) Add(params AddParams) error {
+	ctx := context.Background()
```

Comment:
tiny improvement I think it would be to add the context as a field in App to avoid creating new contexts in each method.

Reply:
That's not recommended you can take a look why here: https://blog.golang.org/context-and-structs

Reply:
You are right, it seems this is not the right approach. The actual goal I was trying to achieve here is to have the same context object instance instead of creating one inside each function like this, to take the most advantage of it. From the post `...of its extensibility in order to build a powerful tree of cancellation, deadline, and metadata information through the call stack...`. So one way that might solve this would be to add a field for the context in Add and follow the same pattern everywhere like in the `git` pkg (where we are creating isolated context for each function) for instance and it would help for testing as well.

Reply:
yes, ideally this should be injected in the method call.

Reply:
The common use case for `context.Context` is to trace requests that cascade into other requests. For us, this would might mean:

```
browser client -> API Server -> K8s API
```
Cancelling the browser request could be made to cancel the K8s API request, or tie then together with a trace, or pass auth information throughout the stack.

Since the command-line invocations do not produce/require a context, its OK to create it at the top-level of this service entry point (IMO). `context.Context` has to be created somewhere.

Reply:
> yes, ideally this should be injected in the method call.

It didn't seem appropriate to change the signature in this PR. It would have made the scope much larger. We can do that as we go.

Reply:
you are totally right, but I had the impression you'd call the App service methods at some point from the API, if that's the case it would make sense to have the context in the method params.

Reply:
I'm good with leaving like that until you need to call it from the API.